src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala

```diff
@@ -144,14 +144,26 @@ class UserService(val directoryDAO: DirectoryDAO, val cloudExtensions: CloudExte
       case Some(_) =>
         for {
           _ <- tosService.acceptTosStatus(userId)
-          enabled <- directoryDAO.isEnabled(userId, samRequestContext) 
+          enabled <- directoryDAO.isEnabled(userId, samRequestContext)
```

Comment:
Note: I have no intuition for what `directoryDAO` means.

Reply:
Note: I have no intuition for what `directoryDAO` means.

Reply:
Good question! The `directoryDAO` refers to Sam's Postgres instance and is the source of truth in Sam. The `registrationDAO` is LDAP/OpenDJ which is essentially a cache of enabled users that service Apache proxies check to determine if a user is enabled in Terra or not. 

We disable a user in the `registrationDAO` if they have not accepted the ToS so that the proxies will reject their requests, but we do not disable them in the `directoryDAO`. We reserve disabling users in the `directoryDAO` for cases where we need to quickly revoke a user's access to Terra, typically if they are a malicious user. In those cases, we _also_ disable them in the `registrationDAO`. 

When a user accepts the ToS, we only want to re-enable them in the `registrationDAO` if they are enabled in the `directoryDAO`. Without this check, a malicious user that had been disabled by a Terra admin would effectively be able to re-enable themselves by accepting the ToS. When rejecting the ToS, we don't need the same check as we want to disable the user in the `registrationDAO` regardless of their enabled/disabled status in the `directoryDAO` as this will prevent them from using Terra until they accept the ToS.

Reply:
Thank you so much-- this was very helpful!